dkim.js

```diff
@@ -53,7 +56,7 @@ function DKIMObject (header, header_idx, cb, timeout) {
     this.sig = header;
     this.sig_md5 = md5(header);
     this.run_cb = false;
-    this.header_idx = header_idx;
+    this.header_idx = cloneHack(header_idx);
```

Comment:
I'd honestly just inline the copy - it's a well understood idiom in JS now for copying.